src/types/delete_folder.rs

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde::Deserialize;
+use xml_struct::XmlSerialize;
+
+use crate::{
+    types::sealed::EnvelopeBodyContents, BaseFolderId, DeleteType, Operation, OperationResponse,
+    ResponseClass, ResponseCode, MESSAGES_NS_URI,
+};
+
+/// A request to delete a folder.
+///
+/// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deletefolder>
+#[derive(Clone, Debug, XmlSerialize)]
+#[xml_struct(default_ns = MESSAGES_NS_URI)]
+pub struct DeleteFolder {
+    /// DeleteFolder uses the DeleteType attribute
```

Comment:
```suggestion
    /// The method the EWS server will use to perform the deletion.
```

suggestion: Both the context in code and in generated documentation give us the context for determining types, but it's harder to tell what effect setting the field has. This wording is copied directly from `DeleteItem`. Additionally, doc comments should conventionally end with a period (with the exception of the links to external documentation).